### ISSUE TITLE
`Rule Request: [Only allow trailing commas for multiline arrays]`

### ISSUE THREAD

**Issue by SylvainRX** [realm/SwiftLint#2653]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Since Swift allows it, having a trailing comma in a multiline array makes it easier to move lines from the bottom to the top or the inverse.
2. Triggers: 
<pre>
let a = [1, 2, 3]      // Should not trigger a warning
let b = [1, 2, 3,]     // Should trigger a warning
let c = [1,            // Should not trigger a warning
         2,
         3
]
let e = [1,            // Should not trigger a warning
         2,
         3,
         ]
</pre>
3. The rule should not be enabled by default because it would have conflicts with others like "trailing_commas".

I have been looking for this feature for quite some time now and I couldn't find any configuration matching this. Thanks in advance for your help if you know any way to achieve this with SwiftLint!

**Comment by marcelofabri** [comment]
You should be able to use `trailling_comma` with `mandatory_comma: true`:

```yml
trailling_comma:
  mandatory_comma: true
```

**Comment by SylvainRX** [comment]
I wanted something a bit more permissive so the `c` would not trigger a warning, but that'll do for now. 
Thank you for your answer :)

**Comment by lordzsolt** [comment]
@SylvainRX The reason to require trailing comma is:
- When adding a new element, git diff is clean for the previous line
- You can easily add/remove elements

So `c` should also trigger.
